Google.Api.Gax.Grpc/GrpcCoreChannelFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * Copyright 2020 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+
+namespace Google.Api.Gax.Grpc
+{
+    internal sealed class GrpcCoreChannelFactory
+    {
+#if CORE_API_ONLY
+        internal ChannelBase CreateChannel(string endpoint, ChannelCredentials credentials, GrpcChannelOptions options) =>
+            throw new NotImplementedException("GrpcCoreChannelFactory is not implemented in CORE_API_ONLY builds");
+#else
+        internal ChannelBase CreateChannel(string endpoint, ChannelCredentials credentials, GrpcChannelOptions options) =>
+            // TODO: Cache the result of ConvertOptions
+            new Channel(endpoint, credentials, ConvertOptions(options));
+
+        private List<ChannelOption> ConvertOptions(GrpcChannelOptions options)
+        {
+            List<ChannelOption> ret = new List<ChannelOption>();
+            if (options.EnableServiceConfigResolution is bool enableServiceConfigResolution)
+            {
+                ret.Add(new ChannelOption("grpc.service_config_disable_resolution", enableServiceConfigResolution ? 1 : 0));
```

Comment:
is the 0 : 1 reversed here? (disable vs enable semantics)

Reply:
Whoops - that's just me failing. Fixing.